### PR TITLE
Fixed heap memory increase

### PR DESCRIPTION
**Pull request by cborla**
|Related issue|
|---|
|#17575|

## Description
An increase in the use of memory by the Engine without any justification has been detected, the objective of this Issue is to investigate the cause of this and propose a possible solution.

## DoD
 - [x] Reproduce the memory leak in local, testing it in main branch #11334 . 
 - [x] Memory leak analysis.
 - [x] Porpose a code fix.
 - [x] Check that all test run on green.

### CODE REVIEW

**Comment by cborla** [comment]
## Reproduce the memory leak
In this case the memory usage is increased by rapidjson.
- Branch: [11334-dev-new-wazuh-engine](https://github.com/wazuh/wazuh/tree/11334-dev-new-wazuh-engine)

11334_leak

**Comment by cborla** [comment]
## Tets with proposed fix.
In this case, an increase in memory is observed due to rapidjson.
- Branch: [17575-engine-metrics-memory-leak](https://github.com/wazuh/wazuh/tree/17575-engine-metrics-memory-leak)

11334_no_leak

File: src/engine/source/metrics/src/dataHub.cpp [hunk 1]
```diff
@@ -25,7 +25,8 @@ json::Json DataHub::getResource(const std::string& scope)
 void DataHub::setResource(const std::string& scope, json::Json object)
 {
     const std::lock_guard<std::mutex> lock(m_mutex);
-    m_resources[scope] = object;
+    m_resources.erase(scope);
+    m_resources.insert(std::make_pair(scope, object));
```
Comment (JorgeMarinoDev) [inline src/engine/source/metrics/src/dataHub.cpp:29]: Please check if we do need both lines (erase and insert). Try without erase.

Comment (JorgeMarinoDev) [reply src/engine/source/metrics/src/dataHub.cpp:29]: [] operator from std::map triggers this leak.

**Review by JorgeMarinoDev — APPROVED** [review]
LGTM now. Thank you.

**Review by nmkoremblum — APPROVED** [review]

File: src/engine/source/server/src/endpoints/unixStream.cpp [hunk 2]
```diff
@@ -284,12 +297,18 @@ void UnixStream::createAndEnqueueTask(std::shared_ptr<uvw::PipeHandle> client,
...
             m_metric.m_queueSize->recordValue(m_currentTaskQueueSize.load());
+
             // Check if client is closed
-            if (client->closing())
+            auto client = wClient.lock();
+            if (!client)
+            {
+                LOG_DEBUG("[Endpoint: {}] endpoint: Client already closed (remote close), discarting response", m_address);
+                return;
+            } else  if (client->closing())
```
Comment (nmkoremblum) [inline src/engine/source/server/src/endpoints/unixStream.cpp:311]: Please, fix style here

**Review by nmkoremblum — CHANGES_REQUESTED** [review]

File: src/engine/source/cmds/src/start.cpp [hunk 3]
```diff
@@ -83,6 +87,9 @@ namespace cmd::server
 {
 void runStart(ConfHandler confManager)
 {
+    // exit handler
+    cmd::details::StackExecutor exitHanlder {};
```
Comment (JorgeMarinoDev) [inline src/engine/source/cmds/src/start.cpp:91]: small fix typo exitHanlder  to exitHandler

**Review by JorgeMarinoDev — APPROVED** [review]
LGTM. Just small typo suggested change.

**Review by nmkoremblum — APPROVED** [review]
LGTM

**Review by LucioDonda — APPROVED** [review]
Minor changes in comments but it LGTM!

**Pull request merged**